x-pack/plugins/fleet/server/services/epm/elasticsearch/template/template.ts

```diff
@@ -182,8 +190,8 @@ export function generateMappings(
           dynamicTemplate.path_match = dynamicMapping.pathMatch;
         }
 
-        dynamicTemplateNames.add(name);
-        dynamicTemplates.push({ [dynamicMapping.path]: dynamicTemplate });
+        const size = dynamicTemplates.push({ [name]: dynamicTemplate });
```

Comment:
What does `size` mean in this context? Should it be `index` instead?

Reply:
`push` returns the current size or length of the array, thus the index here is the `size` minus one.